### DISCUSSION TITLE
Create an option form without relationship on a Select field

### DISCUSSION THREAD

**Discussion by nillegal** [filamentphp/filament#5379] [Help] [answered]
I would like to know if it is possible to create an option form on a Select field without a relationship. 
For example,
I have `Vehicle::class` and `Tire::class` .
The `Tire::class` would only returns all the tires stored in the database.
When creating a new `Vehicle::class` , I want to store tires references for each positions as json 
`{"front_left" : "reference", "front_right" : "reference", "rear_left" : "reference", "rear_right" : "reference"}`

If the tire doesn't exist in as a `Tire::class` in the Select field, I want to insert new `Tire::class` reference without linking it to the `Vehicle::class` through a relationship

**Comment by danharrin** [comment, marked as answer]
If you write the logic for adding stuff to JSON in `createOptionUsing()` sure. But I wouldn't recommend it. Store them in the database instead.

**Comment by samueldhuston97** [comment]
Were you able to resolve this issue? If so, how?

**Reply by nillegal** [reply]
Hi,
I actually switched to a very strange approach by assigning default value for each newly created `Car::class`. So I won't have to specify the current `Tire::class` when creating a new `Car::class`. 
I will only have to specify newly mounted `Tire::class` when making an `Intervention::class` which `belongsTo()` a `Car::class`. 

I didn't "really" achieve the initial approach, but at least it could work for now.

Maybe @danharrin could send us a link for `createOptionUsing()` ? We could have missed something else. 🙏🏽

**Reply by samueldhuston97** [reply]
Got it. I'm not sure if your original version was the exact same scenario as ours, but we ended up with the following utilizing @danharrin 's `createOptionUsing()` suggestion.

Select::make('value')->label('Tire')
->createOptionForm([
TextInput::make('name')
->required()
])->createOptionUsing(function ($data){
$tire = new Tire(['name' => $data['name']]);
$tire->save();
return $tire->id;
})
->options(function () {
return Tire::all()->pluck('name', 'id');
})
->searchable(),

This allowed us to save the new "Tire" record to the "Tire" table and then have the "Car" reference the same "Tire" record.

**Reply by nillegal** [reply]
Yes this is exactly what I wanted to do 🙏🏽  appreciate
